enhancements/machine-config/manage-boot-images.md

```diff
@@ -282,44 +285,108 @@ spec:
+	// details is a list of boot image history entries of the machine resource.
+	// +optional
+	Details []BootImageHistoryDetail `json:"details,omitempty"`
 }
 
-// MachineSetBootImageHistoryStatus defines the observed state of MachineSetBootImageHistory
-type MachineSetBootImageHistoryStatus struct {
+type MachineResourceReference struct {
+	// name is the machine management resource's name
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// kind is the machine management resource's kind
+	// +kubebuilder:validation:Required
+	// +required
+	Kind string `json:"kind"`
+	// apiGroup is name of the APIGroup that the machine management resource belongs to. This is for disambiguating
+	// between Cluster API and Machine API backed resources.
+	// +kubebuilder:validation:Required
+	// +required
+	APIGroup string `json:"apiGroup"`
 }
 
 // BootImageHistoryDetail is the struct for each element in the Details array
 type BootImageHistoryDetail struct {
-	Index          int         `json:"index"`
-	UpdatedTime    metav1.Time `json:"updatedTime"`
-	BootImageRef   string      `json:"bootImageRef"`
+	// updateTime records the timestamp at which the update took place.
+	// +required
+	UpdateTime metav1.Time `json:"updatedTime"`
+	// bootImageRef records the new boot image reference to which the update took place.
+	// +required
+	BootImageRef string `json:"bootImageRef"`
 }
 
-// MachineSetBootImageHistoryList contains a list of MachineSetBootImageHistory
-type MachineSetBootImageHistoryList struct {
+// BootImageHistoryList contains a list of BootImageHistory
+type BootImageHistoryList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []MachineSetBootImageHistory `json:"items"`
+	Items           []BootImageHistory `json:"items"`
 }
+
 ```
-There will be one instance of this per `Machineset`. It will be updated by the MSBIC as `Machinesets` are created/updated and will exist in the same namespace as the `MachineSet`. This CRD will also need to support MAPI and CAPI backed `MachineSets`. The goal of this is to provide information about the "lineage" of a `MachineSet` to the user. The user can then manually restore their `MachineSet` to an earlier state if they wish to do so by following documentation. 
+There will be one instance of this per machine management resource(which can be a MachineSet[MAPI or CAPI], MachineDeployment...etc). It will be named the same as the resource being tracked. The MSBIC is responsible for creating and updating this CR when a boot image update takes place. This CR will exist in the same namespace as the resource.
+
+YAML Example for a MAPI backed machineset scenario:
+```
+apiVersion: machineconfiguration.openshift.io/v1alpha1
+kind: BootImageHistory
+metadata:
+  name: djoshy10-2tcqv-worker-a
+spec: {}
+status:
+  machineResourceReference:
+    name: djoshy10-2tcqv-worker-a
+    kind: MachineSet
+    apiGroup: cluster.x-k8s.io/v1alpha3
```

Comment:
You should not include the version, `apiGroup` is just the group, and you don't need the version. You can find the preferred version based on a resource/group mapping